### DISCUSSION TITLE
dynamic imports

### DISCUSSION THREAD

**Discussion by transfluxus** [parcel-bundler/parcel#8637] [Q&A]
Hi,

I am opening up a question, that has been asked before many times, but I have not seen any solution for parcel 2.7 that works.

I cannot use the import function e.g.
```
import("https://cdn.skypack.dev/three@0.130.0")
```
the build fails with
```
🚨 Build failed.

@parcel/core: Failed to resolve 'https://cdn.skypack.dev/three@0.130.0' from './src/models.ts'

  /web/src/models.ts:411:24
  > 411 |                 import("https://cdn.skypack.dev/three@0.130.0")
  >     |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

@parcel/resolver-default: Unknown url scheme or pipeline 'https:'

```

how can I import external js dynamically?!?

thanks

**Comment by transfluxus** [comment]
the most evil solution I figured out:  const importPromise = eval("import(uri)")


webpack has magic comments:   import(/* webpackIgnore: true */ 'http://example.com/some-module/some-module.bundle.js').then(module => console.log(module.default))

Is there something similar work parcel?

**Reply by mischnic** [reply]
A less evil solution is something like `var url = "example.com/...."; import(url)`. Which should be enough to trick Parcel. `eval` is going to cause you some more problems with Terser.

**Reply by transfluxus** [reply]
that doesnt solve it :/ same error
